numpy/core/src/private/ufunc_override.c

```diff
@@ -67,64 +124,52 @@ PyUFunc_WithOverride(PyObject *args, PyObject *kwds,
                      PyObject **with_override, PyObject **methods)
 {
     int i;
-
-    int nargs;
-    int nout_kwd = 0;
-    int out_kwd_is_tuple = 0;
     int num_override_args = 0;
+    int nin, nout = 0;
+    PyObject *out_kwd_obj;
+    PyObject **in_objs, **out_objs;
 
-    PyObject *obj;
-    PyObject *out_kwd_obj = NULL;
-    /*
-     * Check inputs
-     */
-    if (!PyTuple_Check(args)) {
-        PyErr_SetString(PyExc_TypeError,
-                        "Internal Numpy error: call to PyUFunc_HasOverride "
-                        "with non-tuple");
-        goto fail;
+    nin = PyTuple_Size(args);
```

Comment:
This is misleading - it's not the number of inputs, but the number of positional args, some of which may be outputs

Reply:
Actually, I'm not sure if that's true - if it's not, can you clarify that `PyUFunc_WithOverride` never receives out arguments positionally?

Reply:
You are right, it is just the positional arguments. This means that for reductions arguments like `axis` get checked too, which is not great, but out of scope here (there is an issue, #9107, and I do hope to get to it...)

Reply:
Anyway, now clarified this and called things `arg` instead of `in`.